gliderpy/plotting.py

```diff
@@ -90,3 +90,45 @@ def plot_transect(
     ax.set_ylim(ax.get_ylim()[0], 0)
 
     return fig, ax
+
+
+@register_dataframe_method
+def plot_ctd(
+    df: pd.DataFrame,
+    profile_number: int,
+    var: str,
+    ax: plt.Axes = None,
+    color: str | None = None,
+) -> tuple:
+    """Make a CTD profile plot of pressure vs property
+    depending on what variable was chosen.
+
+    :param profile_number: profile number of CTD
+    :param var: variable to plot against pressure
+    :param ax: existing axis to plot on (default: None)
+    :param color: color for the plot line (default: None)
+    :return: figure, axes
+    """
+    g = df.groupby(["longitude", "latitude"])
+    profile = g.get_group(list(g.groups)[profile_number])
+
+    if ax is None:
+        fig, ax1 = plt.subplots(figsize=(5, 6))
+        ax1.plot(profile[var], profile["pressure"], label=var, color=color)
+        ax1.set_ylabel("Pressure")
+        ax1.set_xlabel(var)
+        ax1.legend()
+        ax1.invert_yaxis()
+        return fig, ax1
+
+    fig = ax.get_figure()
+    ax2 = ax.twiny()  # Create a new twinned axis
+    ax2.plot(profile[var], profile["pressure"], label=var, color=color)
+    ax2.set_xlabel(var)
+
+    # Handle legends
+    lines, labels = ax.get_legend_handles_labels()
+    lines2, labels2 = ax2.get_legend_handles_labels()
+    ax.legend(lines + lines2, labels + labels2, loc="lower center")
+
+    return fig, ax2
```

Comment:
Note that you also need to regenerate the figures when you change them. Otherwise you will be comparing a new modified image to the old one.

```suggestion
    return fig, ax
```

Reply:
You're correct . I'm changing it and doing the push again.